be/src/exprs/time_functions.cpp

```diff
@@ -2597,6 +2597,14 @@ StatusOr<ColumnPtr> TimeFunctions::datediff(FunctionContext* context, const Colu
             result.append(l.diff_microsecond(r) / USECS_PER_MILLIS);
         } else if (type_str == "day") {
             result.append(l.diff_microsecond(r) / USECS_PER_DAY);
+        } else if (type_str == "week") {
+            result.append(l.diff_microsecond(r) / USECS_PER_WEEK);
+        } else if (type_str == "quarter") {
+            result.append(months_diffImpl::apply<TimestampValue, TimestampValue, int>(l, r) / 3);
```

Comment:
the current implementation of months_diff didn't consider microseconds, need to add more test caes to confim that if we need do some change to meet our requirements